### ISSUE TITLE
Cannot resolve warnings (intersecting ways)

### ISSUE THREAD

**Issue by cbeddow** [facebook/Rapid#1050]
### Description

![image](https://github.com/facebook/Rapid/assets/10554922/86b81e0a-9557-491b-b694-bf0655e48475)

None of these warnings when I go to submit a changeset can be resolved

They all have the but to click  "connect the features" and I click it and nothing happens

On the map, they all look to actually be resolved

### Version

2.0.3

### What browser are you seeing the problem on? What version are you running?

Chrome v115.0

### The OS you're using

win

### The browser URL at the time you encountered the bug

https://rapideditor.org/edit#background=EsriWorldImageryClarity&datasets=fbRoads,msBuildings&disable_features=boundaries&map=22.64/32.71873/-117.16396&photo_overlay=mapillary

### The auto-detected useragent string for your browser (leave blank if you're manually filling this form out)

Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36

**Comment by cbeddow** [comment]
Graph.js:96 Uncaught Error: Entity n9485242671 not found
    at Dr.entity (Graph.js:96:13)
    at almost_junction.js:195:28
    at Array.forEach (<anonymous>)
    at x (almost_junction.js:193:15)
    at l (almost_junction.js:46:33)
    at N (validator.js:591:22)
    at Array.forEach (<anonymous>)
    at w (validator.js:579:25)
    at validator.js:693:24
    at validator.js:762:32

**Comment by Bonkles** [comment]
Opening the 2.0.3 code in the browser (with source maps) gets me to the following line in almost_junction.js: 

```
    function findConnectableEndNodesByExtension(way) {
      let results = [];
      if (way.isClosed()) return results;

      let testNodes;
      const indices = [0, way.nodes.length - 1];
      indices.forEach(nodeIndex => {
        const nodeID = way.nodes[nodeIndex];
        const node = graph.entity(nodeID);         <-- This guy 
```

**Comment by Bonkles** [comment]
Okay, so the whole issue here was : we reworked/rethought the way we saved changes to the local base graph in the application storage. This meant that certain features in the base graph were NOT getting saved. 

If you edited pre-existing features, refreshed, and then clicked 'restore my changes' really fast (before the OSM data could load) then it could result in this graph error occuring, which could cause undefined UI behavior. 

Strongly suspect that this is the underlying cause for #1048 and #1049
